llvm/lib/Transforms/Scalar/GVN.cpp

```diff
@@ -596,10 +617,37 @@ uint32_t GVNPass::ValueTable::lookupOrAddCall(CallInst *C) {
     return v;
   }
 
+  if (MSSA && AA->onlyReadsMemory(C)) {
```

Comment:
Think this (and the one below) should check isMemorySSAEnabled too as MSSA and MemDep may be both enabled.

Reply:
Updated, thanks!